crates/sui-graphql-rpc/src/types/sui_address.rs

```diff
@@ -6,7 +6,48 @@ use serde::{Deserialize, Serialize};
 
 const SUI_ADDRESS_LENGTH: usize = 32;
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub(crate) struct SuiAddress([u8; SUI_ADDRESS_LENGTH]);
 
-scalar!(SuiAddress, "SuiAddress", "Representation of Sui Addresses");
+#[Scalar]
+impl ScalarType for SuiAddress {
+    fn parse(value: Value) -> InputValueResult<Self> {
+        match value {
+            Value::String(mut s) => {
+                if s.starts_with("0x") {
+                    s = s[2..].to_string();
+                } else {
+                    return Err(InputValueError::custom(
+                        "Invalid SuiAddress. Missing 0x prefix",
+                    ));
+                }
+
+                let bytes = hex::decode(s)?;
+                if bytes.len() != SUI_ADDRESS_LENGTH {
+                    return Err(InputValueError::custom(format!(
+                        "Invalid SuiAddress length: {}",
+                        bytes.len()
+                    )));
+                }
+                let mut arr = [0u8; SUI_ADDRESS_LENGTH];
+                arr.copy_from_slice(&bytes);
+                Ok(SuiAddress(arr))
+            }
+            _ => Err(InputValueError::custom("Invalid SuiAddress")),
+        }
+    }
+
+    fn to_value(&self) -> Value {
+        Value::String(hex::encode(self.0))
```

Comment:
We should add tests for all these new scalar types (that they parse and output as they expect).  For example that `SuiAddress` accepts non-padded addresses, and outputs padded ones (which I believe this implementation does do).